text/0000-portals.md

```diff
@@ -0,0 +1,108 @@
+---
+title: portals proposal
+status: DRAFTED
+created_at: 2019-12-23
+updated_at: 2019-12-23
+pr:
+---
+
+# Short Title
+Out of tree DOM insertion of elements via portals.
+
+## Summary
+Introduce a mechanism in LWC for inserting and managing an element outside of its natural dom tree.
+
+## Motivation
+
+Position of a node in the DOM determines its stacking context with the component having no 
+ability to change it other than create a new stacking context. 
+
+This becomes problematic when one wants to present a popup/tooltip/dialog/dropdown/notification that needs to appear on top of other content on the screen, possibly limited by a parent with `overflow: hidden`, or a higher `z-index`. A first instinct for going around `overflow: hidden` is to use `position: fixed`, but it's also bound by the stacking context, and may not work as expected.
+
+Having a custom element be inserted out of its natural dom tree hierarchy would enable one to have greater control over the stacking context, allowing a predictalbe popup and modal implementation (including faux-modals that are scoped to tabs and are not per-se global).
+
+## Basic example
+
+The proposal suggests two new APIs:
+1) A way to define where in the DOM the element would be inserted (eg. `lwc:portal-definition` attribute on an element)
+2) A way to define the destination portal of an element (eg. `lwc:portal` attribute on an element)
+
+One non-trivial example of such usage:
+
+```xhtml
+<body>
+  <div lwc:portal-definition="global-modals"></div>
```

Comment:
this is problematic because until now, `lwc:` directives are only useful inside a template files for LWC components. How is this different from using an `id`? it is already guaranteed to be unique (or should be), it must be direct child of body (or should be), and it is easy to find from inside a component (just make a global search for it).

Reply:
good point, I agree, it's better to just go with global ids.